selfdrive/logcatd/logcatd_systemd.cc

```diff
@@ -36,17 +36,8 @@ int main(int argc, char *argv[]) {
 
     // Wait for new message if we didn't receive anything
     if (r == 0){
-      do {
-        r = sd_journal_wait(journal, 1000 * 1000);
-        assert(r >= 0);
-      } while (r == SD_JOURNAL_NOP && !do_exit);
-
-      if (do_exit) break;
-
-      r = sd_journal_next(journal);
-      assert(r >= 0);
-
-      if (r == 0) continue; // Try again if we still didn't get anything
+      sd_journal_wait(journal, 1000 * 1000);
```

Comment:
Did you mean?
```suggestion
      r = sd_journal_wait(journal, 1000 * 1000);
```

Reply:
maybe an assert is better.

Reply:
yes. You probably saw my initial comment before I edited it.

Reply:
yes.lol.